### PR TITLE
UIP-49 Fix report fetch test

### PR DESCRIPTION
**Pull request by briehl**
Next in the set of tests to update to pytest + linting is the report fetcher.

This also starts a split between integration tests which need a token to talk to services, and unit tests which are mocked and bypass the NarrativeServiceImpl and NarrativeServiceServer modules.

### CODE REVIEW

File: test/integration/test_reportfetcher.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import pytest
+from installed_clients.FakeObjectsForTestsClient import FakeObjectsForTests
+from installed_clients.KBaseReportClient import KBaseReport
+from installed_clients.WorkspaceClient import Workspace
+from NarrativeService.NarrativeServiceImpl import NarrativeService
+
+
+@pytest.fixture(scope="module")
+def fake_reads(workspace: list[any], fake_obj_for_tests_client: FakeObjectsForTests):
+    """Make a fake reads object and return the UPA for it."""
```
Comment (ialarmedalien) [inline test/integration/test_reportfetcher.py:13]: this is bringing back memories of the SetAPI edits... 😱

File: test/integration/test_reportfetcher.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    assert len(ret["report_upas"]) == 0
+    assert reads_copy_upa == ret["object_upa"]
+
+
+def test_fetch_report_none(
+    context: dict[str, any],
+    service_impl: NarrativeService,
+    fake_report: str
+) -> None:
+    upa = fake_report[0]  # just use the report as an upa. it doesn't have a report, right?
```
Comment (ialarmedalien) [inline test/integration/test_reportfetcher.py:147]: I'm a bit disappointed to think that you can't go into an infinite spiral of reports

File: test/unit/test_sharemanager.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    requester = ShareRequester(params, config)
+    assert isinstance(requester, ShareRequester)
+    assert requester.ws_id == params["ws_id"]
+
+
+invalid_combos = [
+    ({"user": "foo", "share_level": "a"}, "ws_id"),
+    ({"ws_id": 123, "share_level": "a"}, "user"),
+    ({"user": "foo", "ws_id": 123}, "share_level"),
+]
```
Comment (ialarmedalien) [inline test/unit/test_sharemanager.py:27]: does the service check that values are non-None / non-empty?

**Review by ialarmedalien — APPROVED** [review]
I'll leave it to your discretion whether you want to add in the extra tests.

**Comment by briehl** [comment]
There are some other linting fixes in the next PR to that test suite. I'll add the new ones there.

**Pull request merged**